REFRAME.md

```diff
@@ -40,20 +40,57 @@ Given that a client C wants to request information from some server S:
 
 ### HTTP + DAG-JSON
 
-All messages MUST be encoded as DAG-JSON and use explicit content type `application/vnd.ipfs.rpc+dag-json; version=1`
+All messages sent in HTTP body MUST be encoded as DAG-JSON and use explicit content type `application/vnd.ipfs.rpc+dag-json; version=1`
 
+Requests MUST be sent as either:
+- `GET /reframe/{mbase64url-dag-cbor}`
+  - Cacheable HTTP `GET` requests with message passed as DAG-CBOR in HTTP path segment, encoded as URL-safe [`base64url` multibase](https://docs.ipfs.io/concepts/glossary/#base64url) string

```

Comment:
i wonder if base64 of dag-cbor of the query ends up potentially too long (~2k limit iirc) for query paths

Reply:
Correct, this is part of the problem mentioned in option C https://github.com/ipfs/specs/issues/282#issuecomment-1145000892. It probably depends on the request method and what goes into it. This is one of the reasons it might be nice to just have servers handle proper caching of POST requests on their own.

Reply:
Honestly, DAG-JSON, as PeerIDs, Multiaddrs are not in their text representation, but `bytes` anyway..:

```json
{"FindProvidersRequest":{"Key":{"/":"bafybeigvgzoolc3drupxhlevdp2ugqcrbcsqfmcek2zxiw5wctk3xjpjwy"}}}

{"FindProvidersResponse":{"Providers":[{"Node":{"peer":{"ID":{"/":{"bytes":"EiAngCqwSSL46hQ5+DWaJsZ1SPV2RwrqwID/OEuj5Rdgqw"}},"Multiaddresses":[{"/":{"bytes":"NiJwZWVyLmlwZnMtZWxhc3RpYy1wcm92aWRlci1hd3MuY29tBgu43QM"}}]}},"Proto":[{"2304":{}}]}]}}
```